core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -768,6 +771,26 @@ public PlanOptimizers(
                             // Must run before AddExchanges and after ReplicateSemiJoinInDelete
                             // to avoid temporarily having an invalid plan
                             new DetermineSemiJoinDistributionType(costComparator, taskCountEstimator)))));
+
+            // Must run before DetermineTableScanNodePartitioning because PushJoinIntoTableScan creates new table handles
+            // without useConnectorNodePartitioning set which will cause planner to fail
+            builder.add(new IterativeOptimizer(
+                    ruleStats,
+                    statsCalculator,
+                    estimatedExchangesCostCalculator,
+                    ImmutableSet.<Rule<?>>builder()
+                            .addAll(pushIntoTableScanRulesExceptJoins)
+                            .add(new PushJoinIntoTableScan(metadata))
+                            .build()));
+
+            // DetermineTableScanNodePartitioning is needed to needs to ensure all table handles have proper partitioning determined
+            builder.add(new IterativeOptimizer(
+                    ruleStats,
+                    statsCalculator,
+                    costCalculator,
+                    // Must run before AddExchanges
+                    ImmutableSet.of(new DetermineTableScanNodePartitioning(metadata, nodePartitioningManager, taskCountEstimator))));
```

Comment:
just add `DetermineTableScanNodePartitioning` in same `IterativeOptimizer` as `PushJoinIntoTableScan`

Reply:
Much simpler. Thanks.